pkg/cmd/update/update.go

```diff
@@ -0,0 +1,286 @@
+			streams: userio.NewIOStreamsWithInteractive(
+				os.Stdin,
+				os.Stdout,
+				!nonInteractive,
+			),
+		})
+	} else {
+		timeLeft := (updateInterval - timeSince).Round(time.Second)
+		log.Debug().Msgf("next update check will be in %s", timeLeft)
+	}
+}
+
+func getCurrentExecutablePath() string {
+	execPath, err := os.Executable()
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to get executable path")
+	}
+
+	absolutePath, err := filepath.Abs(execPath)
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to get absolute executable path")
+	}
+
+	return absolutePath
+}
+
+func UpdateCmd(cfg *config.Config) *cobra.Command {
+	opts := UpdateOpts{
+		githubToken:   "",
+		targetVersion: "",
+	}
+	updateCmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update corectl",
+		Long:  `Update to the latest corectl version.`,
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) > 0 {
+				opts.targetVersion = args[0]
+			}
+			if cfg.IsPersisted() {
+				opts.githubToken = cfg.GitHub.Token.Value
+			}
+
+			nonInteractive, err := cmd.Flags().GetBool("non-interactive")
+			if err != nil {
+				log.Panic().Err(err).Msg("could not get non-interactive flag")
+			}
+
+			opts.streams = userio.NewIOStreamsWithInteractive(
+				os.Stdin,
+				os.Stdout,
+				!nonInteractive,
+			)
+
+			update(opts)
+		},
+	}
+
+	updateCmd.Flags().BoolVar(
+		&opts.skipConfirmation,
+		"skip-confirmation",
+		false,
+		"Auto approve confirmation prompt for update.",
+	)
+
+	return updateCmd
+}
+
+func update(opts UpdateOpts) {
+	if opts.targetVersion != "" {
+		log.Debug().Msgf("target version set to %s", opts.targetVersion)
+	}
+
+	wizard := opts.streams.Wizard("Checking for updates", "Retrieved version metadata")
+	githubClient := github.NewClient(nil)
+	if opts.githubToken != "" {
+		githubClient = githubClient.WithAuthToken(opts.githubToken)
+	}
+
+	var release *github.RepositoryRelease
+	var err error
+	if opts.targetVersion == "" {
+		release, err = git.GetLatestCorectlRelease(githubClient)
+	} else {
+		release, err = git.GetCorectlReleaseByTag(githubClient, opts.targetVersion)
+	}
+	if err != nil {
+		wizard.Abort(err)
+		log.Panic().Err(err)
+	}
+
+	asset, err := git.GetLatestCorectlAsset(release)
+	if err != nil {
+		wizard.Abort(err)
+		log.Panic().Err(err)
+	}
+	log.Debug().Str("current_version", version.Version).Str("remote_version", asset.Version).Msg("comparing versions")
+	if version.Version == asset.Version {
+		wizard.SetCurrentTaskCompletedTitle(fmt.Sprintf("Already running %s release (%v)", opts.targetVersion, version.Version))
+		wizard.Done()
+		return
+	} else {
+		wizard.SetCurrentTaskCompletedTitle(fmt.Sprintf("Update available: %v", asset.Version))
+		wizard.Done()
+	}
+
+	out, err := glamour.Render(asset.Changelog, "dark")
+	if err != nil {
+		wizard.Abort(err)
+		log.Warn().Err(err).Msg("could not render changelog markdown, falling back to plaintext")
+		fmt.Print(asset.Changelog)
```

Comment:
we should use `IOStreams` object for printing things

Reply:
Please, check all other places.
There are usages in root.go, update.go and version.go files